rundeckapp/grails-app/init/rundeckapp/init/RundeckInitializer.groovy

```diff
@@ -296,12 +298,22 @@ class RundeckInitializer {
             //This is probably a jboss based server
             def vfsFile = location.getContent()
             vfsDirectoryDetected = true
-            if(location.toString().endsWith("WEB-INF/classes")) {
+            if(location.toString().endsWith(WAR_BASE)) {
                 return vfsFile.getPhysicalFile()
             }
         }
 
         try {
+            if(Environment.isWarDeployed()) {
+                String sloc = location.toString()
+                if(sloc.endsWith(".class") && sloc.contains(WAR_BASE)) {
+                    //An unusual case where the location returned by the protection domain code source did not give us a usable location.
+                    //Seen in Tomcat 7 on Centos 7
+                    String webinfclassesdir = sloc.substring(5,sloc.indexOf(WAR_BASE)+WAR_BASE.length())
```

Comment:
what is the magic `5` ?

Reply:
5 is the 'file:' prefix for a url with a file protocol. I've updated the code to reflect that.